### PR TITLE
Hb 07

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/ua/com/cinema/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private Long id;
+
+    @ManyToOne
+    private User user;
+
+    @OneToMany
+    @JoinTable(name = "orders_tickets",
+            joinColumns = @JoinColumn(name = "order_id"),
+            inverseJoinColumns = @JoinColumn(name = "ticket_id"))
+    private List<Ticket> tickets;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/model/Order.java:31]: one more field is missing

File: src/main/java/ua/com/cinema/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setTickets(shoppingCart.getTickets());
+        shoppingCartService.clear(shoppingCart);
+        return orderDao.add(order);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/service/impl/OrderServiceImpl.java]: don't forget to set the date here
also, if add-order operation is not successful, should we clear the cart?

File: src/main/java/ua/com/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -24,11 +25,17 @@ public void addSession(MovieSession movieSession, User user) {
...
-        shoppingCart.getTickets().add(ticket);
-
-        ticketDao.add(ticket);
-        shoppingCartDao.update(shoppingCart);
+        ShoppingCart shoppingCart;
+        try {
+            shoppingCart = shoppingCartDao.getByUser(user);
+        } catch (DataProcessingException e) {
+            return;
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/service/impl/ShoppingCartServiceImpl.java]: we should not catch runtime exceptions - let's remove this

File: src/main/java/ua/com/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -24,11 +25,17 @@ public void addSession(MovieSession movieSession, User user) {
...
+        if (shoppingCart != null) {
+            shoppingCart.getTickets().add(ticket);
+            ticketDao.add(ticket);
+            shoppingCartDao.update(shoppingCart);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/cinema/service/impl/ShoppingCartServiceImpl.java]: what was wrong with the previous solution?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**